modules/apps/dynamic-data-mapping/dynamic-data-mapping-form-field-type/src/main/java/com/liferay/dynamic/data/mapping/form/field/type/internal/fieldset/FieldSetDDMFormFieldTemplateContextContributor.java

```diff
@@ -78,25 +78,55 @@
 					GetterUtil.getLong(
 						ddmFormField.getProperty("ddmStructureLayoutId"))));
 		}
+		else if (Validator.isNotNull(
+					GetterUtil.getString(
+						ddmFormField.getProperty("nestedFieldNames")))) {
+
+			rowsJSONArray = getRowsJSONArray(nestedFields);
+		}
 
 		return HashMapBuilder.<String, Object>put(
 			"ddmStructureId", ddmFormField.getProperty("ddmStructureId")
 		).put(
 			"ddmStructureLayoutId",
 			ddmFormField.getProperty("ddmStructureLayoutId")
 		).put(
-			"nestedFields",
-			getNestedFields(
-				nestedFields,
-				getNestedFieldNames(
-					GetterUtil.getString(
-						ddmFormField.getProperty("nestedFieldNames")),
-					nestedFields.keySet()))
+			"nestedFields", nestedFields
 		).put(
 			"rows", rowsJSONArray
 		).build();
 	}
 
+	protected JSONObject createRowJSONObject(List<Object> nestedFields) {
+		int columnSize = 12 / nestedFields.size();
+
+		JSONArray columnsJSONArray = jsonFactory.createJSONArray();
+
+		for (Object nestedFieldContext : nestedFields) {
+			JSONArray fieldsJSONArray = jsonFactory.createJSONArray();
+
+			fieldsJSONArray.put(
+				MapUtil.getString(
+					(Map<String, ?>)nestedFieldContext, "fieldName"));
+
+			JSONObject columnJSONObject = jsonFactory.createJSONObject();
+
+			columnJSONObject.put(
+				"fields", fieldsJSONArray
+			).put(
+				"size", columnSize
```

Comment:
Don't create the variable columnSize, add the expression inline instead

Reply:
even inside a loop?